app/plugins/Overview/js/overview.js

```diff
@@ -3,25 +3,28 @@
 const IPC = require('ipc');
 // Library for arbitrary precision in numbers
 const BigNumber = require('bignumber.js');
+// Ensure precision
+BigNumber.config({ DECIMAL_PLACES: 24 });
+BigNumber.config({ EXPONENTIAL_AT: 1e+9 });
 // Keeps track of if the view is shown
 var updating;
-// Keeps track of if listeners were already instantiated
-var listening = false;
 
-// Call API and listen for response to call
-function callAPI(call, callback) {
-	IPC.sendToHost('api-call', call);
-	// prevents adding duplicate listeners
-	if (!listening) {
-		IPC.on(call, function(err, result) {
-			if (err) {
-				console.error(err);
-			} else if (result) {
-				callback(result);
-			} else {
-				console.error('Unknown occurence: no error and no result from callAPI!');
-			}
-		});
+// Make API calls, sending a channel name to listen for responses
+function update() {
+	IPC.sendToHost('api-call', '/wallet/status', 'balance');
+	IPC.sendToHost('api-call', '/gateway/status', 'peers');
+	IPC.sendToHost('api-call', '/consensus/status', 'height');
```

Comment:
I'm going to stick to hyphen separated terms like 'new-address' or 'balance-update' instead of camelCase because that's generally a variable naming convention.